packages/library-legacy-sham/src/index.ts

```diff
@@ -1,3 +1,288 @@
+/**
+ * Here we re-export all of the types that we haven't provided sham interfaces for. In doing so,
+ * consumers' projects will typecheck as though they're using the real legacy library, but they may
+ * not work if they also depend on the implementation that underlies the type.
+ */
+export type {
+    Account,
+    AccountBalancePair,
+    AccountChangeCallback,
+    AccountInfo,
+    AccountKeysFromLookups,
+    AccountMeta,
+    AccountsModeBlockResponse,
+    AddressLookupTableAccount,
+    AddressLookupTableAccountArgs,
+    AddressLookupTableInstruction,
+    AddressLookupTableProgram,
+    AddressLookupTableState,
+    AdvanceNonceParams,
+    AllocateParams,
+    AllocateWithSeedParams,
+    AssignParams,
+    AssignWithSeedParams,
+    Authorized,
+    AuthorizedVoter,
+    AuthorizeNonceParams,
+    AuthorizeStakeParams,
+    AuthorizeVoteParams,
+    AuthorizeVoteWithSeedParams,
+    AuthorizeWithSeedStakeParams,
+    BaseTransactionConfirmationStrategy,
+    Blockhash,
+    BlockhashWithExpiryBlockHeight,
+    BlockheightBasedTransactionConfirmationStrategy,
+    BlockProduction,
+    BlockResponse,
+    BlockSignatures,
+    BlockTimestamp,
+    BPF_LOADER_DEPRECATED_PROGRAM_ID,
+    BPF_LOADER_PROGRAM_ID,
+    BpfLoader,
+    CloseLookupTableParams,
+    Cluster,
+    Commitment,
+    CompiledInnerInstruction,
+    CompiledInstruction,
+    CompileLegacyArgs,
+    CompileV0Args,
+    ComputeBudgetInstruction,
+    ComputeBudgetInstructionType,
+    ComputeBudgetProgram,
+    ConfirmedBlock,
+    ConfirmedSignatureInfo,
+    ConfirmedSignaturesForAddress2Options,
+    ConfirmedTransaction,
+    ConfirmedTransactionMeta,
+    ConfirmOptions,
+    Connection,
```

Comment:
I think it's fair to do that for pure types but maybe not for classes since `new Connection()` here will fail in a very confusing way.

Reply:
Oh, I remove types from this array as I implement the shams for them. `Connection` is what I'm doing today.